packages/eslint-plugin/src/index.ts

```diff
@@ -4,10 +4,7 @@ import { processEnvComputedRule } from './process-env-computed.js'
 import { serviceTypeAnnotations } from './service-type-annotations.js'
 import { unsupportedRouteComponents } from './unsupported-route-components.js'
 
-export const rules: Record<
-  string,
-  ESLintUtils.RuleModule<string, never[], ESLintUtils.RuleListener>
-> = {
+export const rules: Record<string, ESLintUtils.RuleModule<string>> = {
```

Comment:
Type signature change with typescript-eslint v8: all but the first type argument are now unnecessary.